crates/core/tedge_actors/src/actors.rs

```diff
@@ -25,7 +25,24 @@ pub trait Actor: 'static + Send + Sync {
     /// Processing input messages,
     /// updating internal state,
     /// and sending messages to peers.
-    async fn run(&mut self) -> Result<(), RuntimeError>;
+    async fn run(self) -> Result<(), RuntimeError>;
```

Comment:
I was wondering why you didn't go with the simpler fix of `self: Box<Self>` (suggested in that same forum discussion) here instead. I'm assuming you did that to make the `run` method implemented by the actors look more natural and hide the complexity of `Box<Self>` with that blanket impl of `ActorBoxed` below.

Reply:
The existing API works well for our use case, and I really don't like tests polluted with boxing values simply because the actual code uses dynamic dispatch somewhere.

`Box<Self>` is very much an implementation detail here, and I want to be clear about that, particularly given the main change that's actually being made here is to the communication of what `run` means and what assumptions can be made when implementing it, which up until now have been implicit. Receiving `Box<Self>` doesn't tell me anything useful as an implementor of the trait, and because it's pretty uncommon thing, it's likely to cause some confusion among people.